1 спринт 1 проект "С чистого листа"

ele0le opened the pull request:
5 когорта, Олейник Елена, 1 спринт, проект "С чистого листа"

ele0le committed:
для первого ревью первого проекта

ele0le committed:
для первого ревью первого проекта

ele0le committed:
Delete s-chistogo-lista-review directory

ele0le committed:
Delete .github/workflows/index.html

ele0le committed:
Delete .github/workflows/style.css

ele0le committed:
Add files via upload

ele0le committed:
для первого ревью первого проекта

ele0le committed:
Delete style.css

ele0le committed:
Add files via upload

ele0le committed:
Delete fonts/fonts.css

ele0le committed:
исправила после 1 автотеста

ele0le committed:
Delete styles/style.css

ele0le committed:
исправила после 1 автотеста

ele0le committed:
Delete styles/style.css

ele0le committed:
исправила после 2 автотеста

ele0le committed:
Delete index.html

ele0le committed:
исправила после 2 автотеста

ele0le committed:
Delete styles directory

ele0le committed:
исправила после 3 автотеста

ele0le committed:
Delete styles directory

ele0le committed:
исправила после 4 автотеста

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+    <a class="header_logo">
+        <img src="/images/logo.svg" alt="Логотип Вёрстка" class="header_logo-image">
+    </a>
+    <nav class="header_menu">
```
отлично

верное решение использовать тег nav

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>С чистого листа</title>
+    <link rel="stylesheet" href="/fonts/fonts.css">
```
надо исправить

необходимо явно указать явно указать на корень проекта `./` чтобы файл подключился к проекту

ele0le replied:
я проверила - если я указываю на корень проекта ./, то у меня наоборот файл не цепляется. А если указываю, как сейчас "/fonts/fonts.css"  и "/styles/style.css", то все правильно цепляется. Почему так?

ele0le replied:
я создала новый файл fonts как мне предложила программа и скопировала туда содержимое из старого файла, чтобы ссылка ./fonts/fonts.css заработала. не знаю, насколько это правильно.

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>С чистого листа</title>
+    <link rel="stylesheet" href="/fonts/fonts.css">
+    <link rel="stylesheet" href="/styles/style.css">
```
надо исправить

необходимо явно указать явно указать на корень проекта `./` чтобы файл подключился к проекту  https://disk.yandex.com/i/pgkkW-wCjFSx4A

ele0le replied:
когда я указываю на корень проекта, у меня файл наоборот не подключается, т.к. программа его не видит.
![2023-09-23_21-43-34](https://github.com/ele0le/s-chistogo-lista/assets/144122126/0fcbc92c-6b6b-4412-ba32-f8c848874c4e)

ele0le replied:
я создала новый файл style.css как мне предложила программа и скопировала туда все содержимое из старого файла, чтобы заработала ссылка ./styles/style.css. не знаю, насколько это правильно...

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+    <nav class="header_menu">
+        <ul class="header_menu-list">
+            <li class="header_menu-item">
+                <a class="header_menu-item_active" href="#0">Главная</a>
```
надо исправить

- будет лучше назначить один класс для всех ссылок в блоке, и использовать отдельный класс для стилизации активной ссылки , так как любая ссылка может быть активной

ele0le replied:
добавила "header_menu-link" к классу активной ссылки, а в файле со стилями поменяла местами правила - сначала указала правило для "header_menu-link", а потом для "header_menu-item_active", чтобы у первой ссылки не было подчеркивания

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+            <li class="header_menu-item">
+                <a class="header_menu-link" href="#0">О нас</a>
+            </li>
+            <li class="header_menu=item">
```
надо исправить

в наименовании класса допущена опечатка

ele0le replied:
исправила на "header_menu-item" на строчке 24.

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+        </ul>
+    </nav>
+</header>
+<main>
```
Отлично

все контентное содержимое обернуто тегом main

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+</header>
+<main>
+    <section class="cover content-section content-section_theme_dark">
+        <h1 class="cover_title">
```
отлично

верный уровень заголовка

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+        </p>
+        <ul class="cards-list">
+            <li>
+                <article class="card card_theme_dark">
```
Отлично

верное решение использовать тег article

yana-yanchenko commented on index.html:
```diff
@@ -0,0 +1,177 @@
+    <p class="footer_copy">
+        © VЁRSTKA   
+    </p>
+    <ul class="footer_links">
```
отлично

верное решение использовать тег ul

yana-yanchenko commented on styles/style.css:
```diff
@@ -0,0 +1,379 @@
+    padding: 24px 20px;
+    display: flex;
+    width: 360px;
+    height: 132px;
```
надо исправить

Высоту блока не следует ограничивать, так как при добавлении/удалении контента отображение сломается.
Количество и размер контента в блоке может в дальнейшем изменяться, высота должна гибко реагировать.

ele0le replied:
убрала height: 132px;

yana-yanchenko commented on styles/style.css:
```diff
@@ -0,0 +1,379 @@
+    display: flex;
+    flex-direction: column;
+    gap: 8px;
+    border-radius: 30px;
```
надо исправить

для контейнера нет необходимости добавлять скругление

ele0le replied:
убрала border-radius: 30px; из правила .card_author

yana-yanchenko commented on styles/style.css:
```diff
@@ -0,0 +1,379 @@
+    /* фото автора */
+    width: 100px;
+    height: 100px;
+    border-radius: 50%;
```
отлично

верное решение использовать значение 50%

yana-yanchenko commented on styles/style.css:
```diff
@@ -0,0 +1,379 @@
+    /* контейнер для контента карточки */
+    display: flex;
+    flex-direction: column;
+    border-radius: 30px;
```
надо исправить

для контейнера нет необходимости добавлять скругление

ele0le replied:
убрала border-radius: 30px; из правила .card_content

yana-yanchenko commented on styles/style.css:
```diff
@@ -0,0 +1,379 @@
+    font-size: 16px;
+    font-weight: 400;
+    line-height: 135%;
+    padding-bottom: 16px;
```
надо исправить

У последнего элемента в группе не должно быть правого отступа, для последнего элемента правый отступ нужно обнулить. Для обращения к последнему элементу в группе воспользуйтесь псевдоэлементом `:last-child`

Подробнее => https://developer.mozilla.org/ru/docs/Web/CSS/:last-child

ele0le replied:
добавила li:last-child {
padding-right: 0;
margin-right: 0;
padding-bottom: 0;
}

yana-yanchenko reviewed (commented):
Добрый день. Работа выполнена на хорошем уровне!

Отлично:

- все комментарии в коде Отлично
- Используются семантически верные теги
- Размечена вся структура

Надо исправить:


- Все комментарии с пометкой "надо исправить"
- необходимо проконтролировать текстовые параметры по макету для текстовых элементов
- В проекте есть несоответствие макету
    - https://disk.yandex.com/i/5AnkWUo8JZi2MQ
    - https://disk.yandex.com/i/2WTgT-A3kJuAQg
    - https://disk.yandex.com/i/Edl28fsVJ9rIAQ
    - https://disk.yandex.com/i/seaUIc9HCElCTA
    - https://disk.yandex.com/i/v36mEHyCWfpnJg

Можно лучше:

- Все комментарии "Можно лучше" в коде.
- Для ссылок можно добавить эффект наведения с изменением прозрачности (opacity) https://developer.mozilla.org/ru/docs/Web/CSS/opacity
- В проекте верным решением будет использовать БЭМ https://ru.bem.info/methodology/quick-start/ , так же можно использовая валидатор для самоконтроля БЭМ https://nglazov.github.io/bem-validator-page/

Необходимо исправить все "надо исправить". Исправление "можно лучше" остаются на ваше усмотрение.

ele0le committed:
исправила в соответствии с комментариями ревьюера 

после 1 ревью

yana-yanchenko commented on fonts/fonts.css:
```diff
@@ -0,0 +1,35 @@
+@font-face {
+    src: url(./fonts/ebgaramondmedium.woff2) format(".woff2");
```
Надо исправить для каждого подключения : 

- в формате нет необходимости указывать точку , ее нужно удалить
- нет необходимости дополнительно выходить из папки `./fonts`, так как шрифты хранятся на одном уровне с файлом `fonts.css`

ele0le replied:
сделала

yana-yanchenko reviewed (commented):
Добрый день. Отличные исправления, работа принята! Обратите внимание в работе остались комментарии обязательные к исправлению, их необходимо подправить самостоятельно , осталось исправить:

- Все комментарии с пометкой "надо исправить"
- в проекте не отображаются изображения
    - https://disk.yandex.com/i/ZxtsnwPa57WEmw
    - https://disk.yandex.com/i/b5zsZdwV26bpgQ
    - https://disk.yandex.com/i/bLDgvEhzii8okA
- в проекте есть расхождение с макетом 
    - https://disk.yandex.com/i/2N1c3ZSzRFLVzA
    - https://disk.yandex.com/i/KB4aeHctfWx5Fw
- используется ```font-family: 'шрифт'```, это является не совсем корректным решением. 
 В том случае, если по какой-либо причине шрифты не загрузятся или будет использоваться браузер, не поддерживающий текущий формат шрифта - следует предусмотреть такой исход и добавить системные шрифты. 
 Например, Arial, sans-serif

ele0le committed:
исправила второй раз после комментариев ревьюера

выровняла надпись "С чистого листа", текст "Начать любыми словами", исправила ссылки для картинок, исправила ссылки в файле fonts

yana-yanchenko reviewed (approved):
.

yana-yanchenko commented on fonts/fonts.css:
```diff
@@ -0,0 +1,35 @@
+@font-face {
```
Утверждено

yana-yanchenko reviewed (approved):
Утверждено

SmilingJey reviewed (approved):
принято

SmilingJey reviewed (approved):
.

SmilingJey reviewed (approved):
.
